src/main/scala/scalismo/mesh/MeshBoundary.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2015 University of Basel, Graphics and Vision Research Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package scalismo.mesh
+
+import breeze.linalg.CSCMatrix
+import scalismo.geometry.Dim
+
+
+/**
+  * MeshBoundary is a property to test if points or a line between two points is on the boundary of the mesh.
+  */
+trait MeshBoundary {
+
+  /**
+    * Check if the point with given index is on the boundary of the underlaying mesh.
+    *
+    * @return True if point is part of the boundary.
+    */
+  def pointIsOnBoundary(id: Int): Boolean
+
+  /**
+    * Check if the line between the two points with given indices is on the boundary of the underlaying mesh.
+    * @note This method does not check weather there exists a line between the two points. You will get false for all
+    *       lines not in the mesh.
+    *
+    * @return True if the line is part of the boundary.
+    */
+  def lineIsOnBoundary(id1: Int, id2: Int): Boolean
+
+}
+
+
+
+/**
+  * The TriangularMeshBoundary can be queried if a triangle has one side in common with the mesh boundary.
+  */
+trait TriangularMeshBoundary extends MeshBoundary {
+
+  /**
+    * Check if the triangle with given index is on the boundary of the underlaying mesh.
+    *
+    * @return True if the triangle has at least one border in common with the boundary.
+    */
+  def triangleIsOnBoundary(id: Int): Boolean
+}
+
+
+
+/**
+  * Implementation of a TriangularMeshBoundary
+  */
+private class BoundaryOfATriangleMesh(private var vertexIsOnBorder: IndexedSeq[Boolean],
+                                      private var lineIsOnBorder: CSCMatrix[Boolean], // NOTE: CSCMatrix is a yet experimental sparse matrix in breeze
+                                      private var triangleIsOnBorder: IndexedSeq[Boolean]) extends TriangularMeshBoundary {
+
+  override def pointIsOnBoundary(id: Int) : Boolean = {
+    return vertexIsOnBorder(id)
+  }
+
+  override def lineIsOnBoundary(id1: Int, id2: Int): Boolean = {
+    return lineIsOnBorder(id1,id2);
```

Comment:
the operation `CSCMatrix(row, col)` is not necessarily very fast for large columns (does a binary search in the column)

maybe consider a `Map[(PointId, PointId), Boolean]` and ensure symmetry through ordering access?